api/hardwaremanagement/v1alpha1/hardwareplugin_types.go

```diff
@@ -0,0 +1,55 @@
+/*
+SPDX-FileCopyrightText: Red Hat
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// HardwarePluginSpec defines the desired state of HardwarePlugin
+type HardwarePluginSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ApiEndpoint is the root URL for the Hardware Plugin.
+	// +kubebuilder:validation:MinLength=1
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Hardware Plugin API Endpoint",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	ApiEndpoint string `json:"apiEndpoint"`
```

Comment:
Since this is expected to be something like `https://my-plugin.example.com:8443` then could we rename this to `apiRoot` to avoid confusing users into thinking that they need to provide some functional API endpoint.